### ISSUE TITLE
R_CONFIG_ACTIVE set to "rsconnect" during publishing

### ISSUE THREAD

**Issue by nick-youngblut** [jjallaire/config#43]
When I publish a Quarto doc via `quarto_publish_doc()` to Posit Connect, `R_CONFIG_ACTIVE` is set to `rsconnect`, so how is one supposed to use the config R package in this situation if `R_CONFIG_ACTIVE` is changed during publishing?

**Comment by andrie** [comment]
The purpose of `config` is to respond to environment variables that could be set to different values on different machines.  Specifically, `config` reacts to the value of `R_CONFIG_ACTIVE`, and this allows you to set configurations for different machines, e.g. in dev or production.

If you deploy your code to a Connect instance, the `R_CONFIG_ACTIVE` has the value `rsconnect`, as you've discovered (and as documented).

In short, `config` doesn't publish or change this value - it responds to this value on your target machine.

How you then use config is going to depend a little on your use case, but you'll probably want to have values set for the case where `config` encounters `rsconnect`.

For example, use:

```
default:
  trials: 5
  dataset: "data-sampled.csv"
  
rsconnect:
  trials: 30
  dataset: "data.csv"
```

**Comment by andrie** [comment]
I've added a vignette for using Connect with config at https://rstudio.github.io/config/articles/rsconnect.html
